### ISSUE TITLE
DCC misbehavior for large files (>4 GiB)

### ISSUE THREAD

**Issue by ArchUser** [hexchat/hexchat#382]
Windows 8 
Version: 2.9.4

I was downloading a 10GB Bluray rip and it got to 10076/10077 and when I restarted the download it started downloading from ~2000. The server supports continuation and it was acting like it was continuing but started at the wrong point and kept adding to file even though it was past the total size of the file. I tried redownloading the file and the same issue happened again. Crashed at 10076/10077 then started downloading from ~1800.

**Comment by optyler** [comment]
Well, same issue here for a 4,8Go file, I had to resume few times because of network/bot connection issues... most of the resumes worked well, and I downloaded about 4,16Go, but I can't finish this download, resume create always a new file `abc.tar.1` instead of resume the `abc.tar` file :(

**Comment by typhoon71** [comment]
Uhm, I'm using HexChat 2.11.0 (x86), downloaded a month ago I think, my OS is windows 7 x64, and I have no issues... 
Are you sure you're using the last build? 
Also, if you're on linux, check that the build in the package is recent.

**Comment by optyler** [comment]
I'm using 2.10.2 on windows 7 x64, where did you download the 2.11? when I click on "check for update" it said i'm already in last version :-/

**Comment by tomek** [comment]
@optyler 2.11 is a test build and you can download it from here: https://dl.hexchat.net/hexchat/testing/

**Comment by typhoon71** [comment]
Truth to be said I did download this build around a month ago, but before that I was still using the build from 2014, when I first posted here. XD

**Comment by optyler** [comment]
well, FYI, after i get installed this nightbuild, I got this error on launch : "api-ms-win-crt-runtime-l1-1-0.dll missing", this is strange, cause the previous version starts well... I didn't launce Microsoft updates for years... maybe I should do an update first :-/

**Comment by typhoon71** [comment]
You need to install the "Visual C++ Redistributable for Visual Studio 2015".

**Comment by Arnavion** [comment]
The installer will install it for you.

**Comment by typhoon71** [comment]
@Arnavion: 
In my case I don't think it installed the runtimes, maybe because my firewall blocked it, so I run in a similar error and I found out it was the need for the runtimes.

Also: on hexchat download page it still points to the 2013 runtimes, shouldn't that be the 2015 ones?

**Comment by Arnavion** [comment]
It links to the runtime needed for the latest release.

**Comment by optyler** [comment]
I tryed to install x64 and x32, update via windows update, install automaticly and manualy the runtimes, but I get the same issue on start... No much time to investigate (Xmas :santa: )
I'll report here future tests ;)

**Comment by illeatmyhat** [comment]
Using HexChat 2.11, I can download >4GB files (tested file was 10GB).
While the file almost always fails midway, I just have to resume it and it works.
I'm not sure if the file failing is due Hexchat or an issue on the other side.

**Comment by KilimIG** [comment]
Me and some friends just finished setting up an xdcc bot using iroffer (dinoex mod) and most of us ran into this problem on the newest public build of Hexchat (2.12.1 x64). We were all trying to download the same file (6.5GB). It would get stuck on random points after 2GB; for me personally it was stuck at 2.4GB, 2.8GB and 3.4GB. Each time was a different problem. Resumes did not function for any of us. 

Interestingly enough there was one successful download with the latest hexchat build, he was on Windows 7. However another user on Windows 7 had a failed download. On top of that, someone using adiirc ran into the exact same problem. There doesn't seem to be any rhyme or reason from this from all the people sampled. is it just a random issue?

tl;dr doesn't work on newest build and resumes not supported.
